src/components/WorkspaceFreeform.js

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Rnd } from 'react-rnd';
+import Window from '../containers/Window';
+import ns from '../config/css-ns';
+
+/**
+ * Represents a work area that contains any number of windows
+ * @memberof Workspace
+ * @private
+ */
+class WorkspaceFreeform extends React.Component {
+  /**
+   */
+  render() {
+    const {
+      workspace,
+      windows,
+      setWorkspaceViewportPosition,
+    } = this.props;
+    return (
+      <Rnd
+        default={{
+          width: 5000,
+          height: 5000,
+        }}
+        position={{ x: workspace.viewportPosition.x, y: workspace.viewportPosition.y }}
+        enableResizing={{
+          top: false,
+          right: false,
+          bottom: false,
+          left: false,
+          topRight: false,
+          bottomRight: false,
+          bottomLeft: false,
+          topLeft: false,
+        }}
+        onDragStop={(e, d) => {
+          setWorkspaceViewportPosition({ x: d.x, y: d.y });
+        }}
+        cancel={`.${ns('window')}`}
+        className={ns('workspace')}
+      >
+        {
+          Object.values(windows).map(window => (
+            <Window
```

Comment:
Could this be a seam to render the `draggableWorkspaceWindow` (perhaps as a component) with a child of `Window`? Could that remove the conditional from `Window`?

Reply:
Yeah, that's probably nicer. There was a lot of reshuffling. I'll make this update.